test/utilities/testsUtils.ts

```diff
@@ -0,0 +1,224 @@
+ *
+ * @param {TreeItem} testItem - The test item whose icon color needs to be verified. It represents a node in the sidebar tree view.
+ * @param {string} colorLabel - The expected color label (e.g., 'testPass', 'testNotRun') that indicates the test status.
+ *
+ * @example
+ * await verifyTestIconColor(myTestItem, 'testPass'); // Verifies the icon is green for a passing test
+ */
+export async function verifyTestIconColor(testItem: TreeItem, colorLabel: string) {
+  log(`Verifying icon's colors - verifyTestIconColor()`);
+  // Verify the tests that are passing are labeled with a green dot on the Test sidebar
+  const icon = await (await testItem.elem).$('.custom-view-tree-node-item-icon');
+  const iconStyle = await icon.getAttribute('style');
+  // Try/catch used to get around arbitrary flaky failure on Ubuntu in remote
+  try {
+    await expect(iconStyle).toContain(colorLabel);
+  } catch {
+    log('ERROR: icon did not turn green after test successfully ran');
+  }
+}
+
+/**
+ * Verifies that the test result contains all expected text snippets.
+ *
+ * @param {string} testResult - The actual test result as a string that needs to be verified.
+ * @param {string[]} expectedTexts - An array of strings representing the expected text snippets that should be present in the test result.
+ *
+ * @example
+ * await verifyTestResult(
+ *   testResult,
+ *   [
+ *     '=== Test Summary',
+ *     'Outcome              Passed',
+ *     'Tests Ran            1',
+ *     'Pass Rate            100%',
+ *     'TEST NAME',
+ *     'ExampleTest1  Pass',
+ *     'ended SFDX: Run Apex Tests'
+ *   ]
+ * );
+ */
+export async function verifyTestResult(testResult: string, expectedTexts: string[]): Promise<void> {
+  for (const expectedText of expectedTexts) {
+    await expect(testResult).toContain(expectedText);
+  }
+}
+
+/**
+ * Verifies the presence of test items in the sidebar.
+ * *
+ * @param {ViewSection} testsSection - An instance of the view section representing the sidebar where test items are displayed.
+ * @param {string} refreshCommand - The command used to refresh the sidebar to ensure it displays up-to-date information.
+ * @param {string[]} expectedItems - An array of strings representing the expected test items that should be present in the sidebar.
+ * @param {number} expectedTests - The expected number of tests to be displayed in the sidebar.
+ * @param {number} expectedClasses - The expected number of test classes to be present in the sidebar.
+ *
+ * @returns {Promise<void>} A promise that resolves when the verification is complete.
+ *
+ * @example
+ * await verifyTestItemsInSideBar(
+ *   mySidebarSection,
+ *   'Refresh Tests',
+ *   ['Test Item 1', 'Test Item 2'],
+ *   2,
+ *   1
+ * );
+ */
+export async function verifyTestItemsInSideBar(
+  testsSection: ViewSection,
+  refreshCommand: string,
+  expectedItems: string[],
+  expectedTests: number,
+  expectedClasses: number
+) {
```

Comment:
```suggestion
): Promise<void> {
```

To get rid of the warning from the 3 dots
![Screenshot 2024-09-09 at 10 15 29 AM](https://github.com/user-attachments/assets/dc419310-473f-449f-b37a-f565da485491)